src/main/java/org/generationcp/middleware/api/file/FileMetadataServiceImpl.java

```diff
@@ -214,11 +214,11 @@ public FileMetadataDTO save(
 		Preconditions.checkArgument(isBlank(observationUnitUUID) != isBlank(germplasmUUID));
 
 		if (!isBlank(observationUnitUUID)) {
-			final ExperimentModel experimentModel = this.daoFactory.getExperimentDao().getByObsUnitId(observationUnitUUID);
-			if (experimentModel == null) {
+			final Optional<ExperimentModel> experimentModelOptional = this.daoFactory.getExperimentDao().getByObsUnitId(observationUnitUUID);
```

Comment:
Implement as common method to validate if `experimentModelOptional` is present:
if (!experimentModelOptional.isPresent()) {
				throw new MiddlewareRequestException("", "filemetadata.observationunit.not.found", new String[] {observationUnitUUID});
			}